src/compute/src/render/context.rs

```diff
@@ -114,6 +121,7 @@ where
             as_of_frontier,
             until: dataflow.until.clone(),
             bindings: BTreeMap::new(),
+            shutdown_token: Weak::new(),
```

Comment:
There's never a case where we create a context like this and don't populate `shutdown_token`, right?  Because it seems like if that happened, the error logging would not log in those cases.  The code currently looks fine, since I guess there must always be at least one object to build, but I'm just wondering if there's any way to enforce that you can't accidentally create an error-muffling context.

Reply:
Yeah, this seems like a potential future pitfall. I'll make this field an `Option<Weak<()>>` instead.